mantidproject/mantid
# Show a tooltip on hovering over the instrument viewer

**Description of work.**
At the request of some scientists at ILL, when hovering over the instrument in the instrument viewer, a tool tip showing basic information about the pixel appears. It is mostly useful and meant for rectangular detectors.

This tool tip only shows the detector name - which in rectangular detectors contains all relevant info about its position - and the integrated counts.

This feature is of course optional and can be activated via the settings menu.


**To test:**
- Load a file with a rectangular detector - for example `ExternalData/Testing/Data/UnitTest/ILL/D22B/046600.nxs`
- Open the instrument viewer.
- In the `Display settings` menu, check `Tooltip`
- Hover over the instrument and check the tool tip is there.



Closes #33567 .




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
